src/lib.rs

```diff
@@ -58,48 +58,68 @@
 //! [`HidDevice`] handles can access the same physical device. For backward compatibility this is
 //! an opt-in that can be enabled with the `macos-shared-device` feature flag.
 
-extern crate libc;
-#[cfg(all(feature = "linux-native", target_os = "linux"))]
-extern crate nix;
-
-#[cfg(target_os = "windows")]
-use windows_sys::core::GUID;
-
 mod error;
 mod ffi;
 
-#[cfg(hidapi)]
-mod hidapi;
-
-#[cfg(all(feature = "linux-native", target_os = "linux"))]
-#[cfg_attr(docsrs, doc(cfg(all(feature = "linux-native", target_os = "linux"))))]
-mod linux_native;
-#[cfg(target_os = "macos")]
-#[cfg_attr(docsrs, doc(cfg(target_os = "macos")))]
-mod macos;
-#[cfg(target_os = "windows")]
-#[cfg_attr(docsrs, doc(cfg(target_os = "windows")))]
-mod windows;
-
-#[cfg(feature = "windows-native")]
-#[cfg_attr(docsrs, doc(cfg(all(feature = "windows-native", target_os = "windows"))))]
-mod windows_native;
-
 use libc::wchar_t;
 use std::ffi::CStr;
 use std::ffi::CString;
 use std::fmt;
 use std::fmt::Debug;
 use std::sync::Mutex;
+use cfg_if::cfg_if;
 
 pub use error::HidError;
 
-#[cfg(hidapi)]
-use crate::hidapi::HidApiBackend;
-#[cfg(all(feature = "linux-native", target_os = "linux"))]
-use linux_native::HidApiBackend;
-#[cfg(all(feature = "windows-native", target_os = "windows"))]
-use windows_native::HidApiBackend;
+cfg_if! {
+    if #[cfg(all(feature = "linux-native", target_os = "linux"))] {
+        //#[cfg_attr(docsrs, doc(cfg(all(feature = "linux-native", target_os = "linux"))))]
```

Comment:
Why comment out these cfg_attrs, and not the ones below?

Reply:
The attribute is used to document that a certain function is limited to certain configurations:
![image](https://github.com/ruabmbua/hidapi-rs/assets/5053369/8c70b94a-a264-49d3-a8b8-119de911a50f)

The native backends should not appear to the public api and therefore the attribute seemed pretty useless. However I just commented them out for now to make it easier to reenable them in case I am wrong and they did actually serve a purpose that I missed.

Reply:
Yeah I think you are right. We can just remove them completely, since its backend code.